module1-introduction-to-backend/r1-introduction-to-backend/README.md

```diff
@@ -0,0 +1,73 @@
+
+### What are clients?
+The clients are entities that send requests to the backend. They are often browsers that make requests for the HTML and JavaScript code that they will execute to display websites to the end user. However, there are many different kinds of clients: they might be a mobile application, an application running on another server, or even a web enabled smart appliance.
+
+### What is a backend?
+The backend is all of the technology required to process the incoming request and generate and send the response to the client. This typically includes three major parts:
+- The server. This is the computer that receives requests and sends responses.
+- The app. This is the application running on the server that processes the requests, retrieves information from the database, and sends a response.
+- The database. Databases are used to organize and persist data.
+
+### What is a server?
+A server is simply a computer that listens for incoming requests. Though there are machines made and optimized for this particular purpose, any computer that is connected to a network can act as a server. In fact, you will often use your very own computer as server when developing apps.
+
+The term "server" is also very often used to refer directly to the backend application running on the server. For example, if we build a backend application to manage the student data of a university, we can refer to it as the student server. When you hear a developer say "the sever has crashed", they could either mean the backend application has crashed due to an error in the code or the server running the application has crashed due to a hardware issue.
+
+Frontends also have servers. Technically when you build a website or a web application, you will be hosting it on a service like Netlify or Heroku, which means it will live on a server. However, each time a new user visits this site from their browser, the server will send the HTML to the user's browser where it will then start running as a client.
+
+Once we start building our backend applications we will observe the use of this term more in action.
+
+### What kinds of responses can a server send?
+The data that the server sends back can come in different forms. For example, a server might serve up an HTML file, send data as JSON, or it might send back only an HTTP status code. You’ve probably seen the status code "404 - Not Found" whenever you’ve tried navigating to a URI that doesn’t exist, but there are many more status codes that indicate what happened when the server received the request.
+
+> Okay I want to see some code now!
+
+We heard you. In the next lesson, we will learn about Node.js which will be the primary server-side programming framework that we will be using throughout this bootcamp.
+
+---
+## References
+- https://www.freecodecamp.org/news/front-end-developer-vs-back-end-developer-definition-and-meaning-in-practice
```

Comment:
Well by references I mean these are the links from where I have taken reference content for these readings, because I haven't written everything from scratch. I'm not recommending that students must also read through all of these links, those further readings are in inline links within the content. That's why I haven't added any text description.